site/source/docs/api_reference/proxying.h.rst

```diff
@@ -0,0 +1,144 @@
+.. _proxying-h:
+
+==========
+proxying.h
+==========
+
+The Emscripten `proxying.h
+<https://github.com/emscripten-core/emscripten/blob/main/system/include/emscripten/proxying.h>`_
+API provides a mechanism for dispatching work to be executed by a target thread
+and optionally blocking until that work has been completed. This is particularly
+useful in a Web context where certain JS APIs can only be called on particular
+threads; any thread can access those APIs by proxying the API calls to the
+correct threads. Proxying is also useful for wrapping asynchronous JS tasks with
+a synchronous interface. The calling thread can proxy the asynchronous task to a
+dedicated worker thread and wait until the worker thread eventually marks the
+task as complete, possibly after returning multiple times to the JS event loop.
+
+Work on proxying queues can be executed in two ways. First, the user may
+explicitly call ``emscripten_proxy_execute_queue`` to execute any tasks that
+have been queued for the current thread. Alternatively, if the target thread
+returns to the JS event loop, for example via
+``emscripten_exit_with_live_runtime``, the proxying queue implementation will
+``postMessage`` the target thread to wake it up and tell it to automatically run
+the enqueued work.
+
+See `test_pthread_proxying.c
+<https://github.com/emscripten-core/emscripten/blob/main/tests/pthread/test_pthread_proxying.c>`_
+and `test_pthread_proxying_cpp.cpp
+<https://github.com/emscripten-core/emscripten/blob/main/tests/pthread/test_pthread_proxying_cpp.cpp>`_
+for examples of how to use the proxying API.
+
+API Reference
+=============
+
+Types
+-----
+
+.. c:type:: em_proxying_queue*
+
+  An opaque handle to a set of thread-local work queues to which work can be
+  asynchronously or synchronously proxied from other threads. When work is
+  proxied to a queue on a particular thread, that thread is notified to start
+  processing work from that queue if it is not already doing so.
```

Comment:
The last sentence seems unnecessary, for similar reasons as the `postMessage` before. Rather than say "the thread is notified" - the mechanism - I think it would be better to say what happens, as observed by the user. Say "Each thread will execute the proxied work sent to it".